pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/SchemaRegistryServiceImpl.java

```diff
@@ -293,12 +295,28 @@ private void checkCompatible(SchemaAndMetadata existingSchema, SchemaData newSch
             SchemaData schemaData) {
         final CompletableFuture<SchemaVersion> completableFuture = new CompletableFuture<>();
         SchemaVersion schemaVersion;
-        for (SchemaAndMetadata schemaAndMetadata : schemaAndMetadataList) {
-            if (Arrays.equals(hashFunction.hashBytes(schemaAndMetadata.schema.getData()).asBytes(),
-                    hashFunction.hashBytes(schemaData.getData()).asBytes())) {
-                schemaVersion = schemaAndMetadata.version;
-                completableFuture.complete(schemaVersion);
-                return completableFuture;
+        if (schemaData.getData().length != 0) {
+            Schema.Parser parser = new Schema.Parser();
+            Schema newSchema = parser.parse(new String(schemaData.getData(), UTF_8));
```

Comment:
are we sure that here we are dealing with an Avro schema ?
what happens in case of a non-Avro schema ?

Reply:
yes, this commit only test any test will not pass and then I will reimplement this pr.

Reply:
very good!

we are still missing PROTOBUF tests, correct ?

Reply:
PROTOBUF also use avro struct to store.

Reply:
Sorry I was not clear.
My comment is more about having code coverage for the condition about PROTOBUF.
if we add an explicit unit test about the fact that with PROTOBUF we are performing your new code, we are sure that we won't lose this implementation in the future.

Probably adding an utility method `isUsingAvroSchemaParser` and add simple unit tests about that method will reduce the number of combinations we need to completely cover this change.

Reply:
good idea, i will add the method ```isUsingAvroSchemaParser``` and add test for it.